### ISSUE TITLE
findQuery with include not quite working

### ISSUE THREAD

**Issue by cowboy** [endpoints/ember-data-endpoints#1]
This is (roughly) my applicants model:

``` js
DS.Model.extend({
  // stuff
  position: DS.belongsTo('position'),
  applicant_state: DS.belongsTo('applicant-state'),
  // more stuff
});
```

Doing `this.store.find('applicant')` in my applicants route makes a request to `http://api.loc/applicants?include=position,applicant_state`

But I need to change `applicant_state` to `status` in the include, because that's how the relationship was defined in the API.

So I did `this.store.findQuery('applicant', {include: 'position,status'})` instead (per @tkellen), but that ended up making a request to `http://api.loc/applicants?include=position,applicant_state&include=position%2Cstatus`

Should the `include` query param be getting replaced? Should I be doing it differently?

FWIW, we've changed the API to work around the issue, so no rush on this.

**Comment by bmac** [comment]
Thanks Ben. I think you are correct when doing `this.store.findQuery('applicant', {include: 'position,status'})` the `include` parameter should be overwritten. I'll try to get to this tonight.

**Comment by bobholt** [comment]
This is the wonkiest point for me, because the API is inconsistent between what it's sending as model properties (`leave_request_id`), and what it's expecting as an include param (`&include=request`).

I'm actually doing the reverse by querying a `hasMany`, but my answer was to set up my models like so:

``` js
//leave-request-activity.model
export default DS.Model.extend({
  leaveRequest: DS.belongsTo('leaveRequest'),
});

// leave-request.model
export default DS.Model.extend({
  activity: DS.hasMany('leave-request-activity')
});
```

I would have preferred the second to be `leaveRequestActivities: DS.hasMany('leave-request-activity')`.

I think it's totally cool to assume `table_name_id`: that's what the ActiveModel does, but the API will have to match that in its internal relations.

If that's undesirable, then the API should send its relation names as the property names, and we should override the keyForRelationship method in the serializer. Either way, the linked data type and the links object should match.

**Comment by tkellen** [comment]
@bobholt can you read this? https://github.com/bocoup/api/issues/145

**Comment by tkellen** [comment]
I think our include properties are correct, the links object states which type of resource it points to under the `type` key. An example of this being needed would be `/books/1?include=chapters,firstChapter`. For this you would get two entries in links, keyed `chapters` and `firstChapter`, but they would both associate with the resource of the type `chapters`.

**Comment by bobholt** [comment]
Responding here, to say I wouldn't expect the API to send `firstChapter`. That API call should be something like `?include=chapters&sort=ascending&limit=1`, and the entries should both be keyed `chapters`.

I'll talk about the main issue over in https://github.com/bocoup/api/issues/145

**Comment by bobholt** [comment]
actually, I see the sort and limit would apply to the book, not chapters, but i'm not thrilled by having two different types of relation of the same type

**Comment by tkellen** [comment]
It happens **all the time**, especially with polymorphic relations.  That's why the json-api spec specifies a links object with a `type` key.

**Comment by cowboy** [comment]
I would SUPER DUPER love it if we could just have the same names everywhere (excepting changes from underscore to hyphen to :dromedary_camel:-case.

@bobholt's inline code example looks like the start of a maintenance nightmare.

**Comment by bobholt** [comment]
And changing `leaveRequest: DS.belongsTo('leaveRequest')` to `request: DS.belongsTo('leaveRequest')`, which is what it looks like i should do for type inference blows everything up, _even though it's not referenced anywhere in my app_, which is making me nuts.

I can see getting all of that data back under the type-inferenced name. If we never looked at api.bocoup.com's models, we'd never know they were called anything else. Cool.

But it's all super-gross on the client side with `activity: DS.belongsTo('leaveRequestActivity')`. I can't name the model anything else, because there could potentially be another type of activity.

I'm leaning toward ripping buildURL out of the adapter, so we can just write:

`leaveRequestActivities: DS.hasMany('leaveRequestActivity')`

and then do `this.store.find('leave-request', {include: 'activity'})` in the route - be explicit about the include and not make the adapter too magic.

It seems like we're doing a lot of workarounds to work with json-api, which still seems not-fully-baked just because three guys in a treehouse wrote it and told everyone else it was a standard. But that's probably Friday afternoon frustration talking.

**Comment by bobholt** [comment]
fixed via #7 - endpoints adapter no longer automatically creates include query string
